### PR TITLE
Client hints support

### PR DESCRIPTION
**Pull request by jasonwaters**
## Description

This PR adds support for client hints.  Client hints can be specified in a `DeliveryRequest` in the `request>context>clientHints` property.  When provided, client hints take precedent over details found in the `userAgent` string.

This is not absolutely necessary for ODD, since even the reduced `userAgent` google intends to support into the forseeable future contains the platform, browser and major browser version.  But in order to have parity with delivery api and keep pace with changes to `userAgent` going forward we have decided to support client hints even if at first it may seem redundant.

Another change that was added is the strings used for target platforms.  I found that the strings used were inconsistent with values that should be in the artifact for platform.  So this PR also updates platform strings generated for ODD context to be one of either ["linux", "windows", "mac"].  These are the only supported platform strings in use by target audiences.

## Related Issues

- https://jira.corp.adobe.com/browse/TNT-43815
- https://jira.corp.adobe.com/browse/TNT-43816

### CODE REVIEW

File: packages/target-tools/src/clientInfo.js [hunk 1]
```diff
@@ -105,3 +112,78 @@ export const deviceTypeFromUserAgent = userAgent =>
...
+        versionGroupIndex: 1,
+        regex: /Chrome";v="([0-9.]+)"/i
+      },
+      {
+        name: "Edge",
+        versionGroupIndex: 1,
+        regex: /Edge";v="([0-9.]+)"/i
+      },
+      {
+        name: "Safari",
```
Comment (shandilya3) [inline packages/target-tools/src/clientInfo.js:135]: Safari & Firefox don't support Client Hints, how do we test for them ?
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-CH

**Review by shandilya3 — APPROVED** [review]

File: packages/target-tools/src/clientInfo.js (hunk 1 above)
Comment (jasonwaters) [reply packages/target-tools/src/clientInfo.js:135]: You're right, these regex were added in anticipation of safari and firefox supporting them.  No harm done if they never do.  May need to refine later.

**Review by dcottingham — APPROVED** [review]
Looks great, just have one comment.

File: packages/target-decisioning-engine/src/contextProvider.js [hunk 2]
```diff
@@ -25,19 +26,36 @@ function getLowerCaseAttributes(obj) {
   return result;
 }
 
+const BROWSER_PLATFORMS_MAPPING = {
+  "Windows": "windows",
+  "Macintosh": "mac",
```
[the remark this answers is not in the document]
Comment (jasonwaters) [reply packages/target-decisioning-engine/src/contextProvider.js:31]: For audience targeting, we only supports 3 platforms ("mac", "windows", and "linux").  There are obviously more platforms out there like iOS, Android, etc.  but those are not supported by target audiences.  For that reason this mapping only maps strings that come from user agent or client hints to the 3 platforms we support.

**Pull request merged**